### PR TITLE
Fixed first_map_only parameter issue

### PR DESCRIPTION
**Pull request by hardesh**
Signed-off-by: Harshal Deshpande <hardesh1deshpande@gmail.com>

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | https://github.com/ros-planning/navigation.ros.org/pull/284 |
| Primary OS tested on | Ubuntu) |
| Robotic platform tested on | gazebo simulation of Turtlebot) |

---

## Description of contribution in a few bullet points

- Fixed first_map_only parameter issue in AMCL

## Description of documentation updates required from your changes

- Updated its documentation in the docs repo. PR is mentioned in the above table

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by hardesh** [comment]
Hi @SteveMacenski,
I've added the parameter in nav2_amcl. But I'm getting a build error in nav2_util
```
In file included from /home/harshal/galactic_fix_ws/src/navigation2/nav2_util/src/costmap.cpp:19:
/home/harshal/galactic_fix_ws/src/navigation2/nav2_util/include/nav2_util/geometry_utils.hpp:25:10: fatal error: tf2_geometry_msgs/tf2_geometry_msgs.hpp: No such file or directory
   25 | #include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"

```

**Comment by hardesh** [comment]
Checked this https://github.com/ros2/geometry2/blob/galactic/tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.h 

It seems there isn't a hpp file?

File: nav2_bringup/params/nav2_params.yaml [hunk 1]
```diff
@@ -38,6 +38,7 @@ amcl:
     z_rand: 0.5
     z_short: 0.05
     scan_topic: scan
+    first_map_only: true
```
Comment (SteveMacenski) [inline nav2_bringup/params/nav2_params.yaml:41]: remove this from the yaml please

**Comment by SteveMacenski** [comment]
Otherwise LGTM, I just don't think setting this in the yaml is a good default profile

File: nav2_amcl/src/amcl_node.cpp [hunk 2]
```diff
@@ -221,6 +221,10 @@ AmclNode::AmclNode(const rclcpp::NodeOptions & options)
   add_parameter(
     "map_topic", rclcpp::ParameterValue("map"),
     "Topic to subscribe to in order to receive the map to localize on");
+  
+  add_parameter(
+    "first_map_only", rclcpp::ParameterValue(true),
```
Comment (SteveMacenski) [inline nav2_amcl/src/amcl_node.cpp:226]: Default to `false`

**Comment by hardesh** [comment]
Addressed the comments. Is there any way to test this on galactic? 

Also, do you use rolling on Ubuntu 20?

**Comment by SteveMacenski** [comment]
This has a linting error according to CI:

```
Diff with 5 lines
--- src/amcl_node.cpp
+++ src/amcl_node.cpp.uncrustify
@@ -224 +224 @@
-  
+
```

**Comment by hardesh** [comment]
`ament_clang_format --reformat` is formatting multiple files. [Documentation](https://navigation.ros.org/contribute/index.html) mentions that you're using ament_cpplint, ament_uncrustify, ament_cppcheck. So I tried using ament_cpplint, but even that is changing multiple files

**Comment by SteveMacenski** [comment]
It's all good -- there's a little bit of wackiness going on right now due to a migration from 20.04 -> 22.04 and the linting profiles were changed.

**Pull request merged**